### PR TITLE
Skip to home screen if terms of service is missing after signing in

### LINKED ISSUE #1032: [ToS] App crashes when ToS are missing from db
**To Reproduce**
1. Delete the `config/tos` document from a working Ground Firebase project.
2. Wipe app data on device and sign back in.

**Expected behavior**
The project list appears.

**Actual behavior**
An exception is thrown:

```
2021-11-09 18:25:05.731 8390-8390/com.google.android.gnd E/MainViewModel: Error loading terms of service from remote db
    com.google.firebase.firestore.FirebaseFirestoreException: PERMISSION_DENIED: Missing or insufficient permissions.
        at com.google.firebase.firestore.util.Util.exceptionFromStatus(Util.java:117)
        at com.google.firebase.firestore.core.EventManager.onError(EventManager.java:166)
        at com.google.firebase.firestore.core.SyncEngine.removeAndCleanupTarget(SyncEngine.java:588)
        at com.google.firebase.firestore.core.SyncEngine.handleRejectedListen(SyncEngine.java:424)
        at com.google.firebase.firestore.core.MemoryComponentProvider$RemoteStoreCallback.handleRejectedListen(MemoryComponentProvider.java:99)
        at com.google.firebase.firestore.remote.RemoteStore.processTargetError(RemoteStore.java:562)
        at com.google.firebase.firestore.remote.RemoteStore.handleWatchChange(RemoteStore.java:446)
        at com.google.firebase.firestore.remote.RemoteStore.access$100(RemoteStore.java:53)
        at com.google.firebase.firestore.remote.RemoteStore$1.onWatchChange(RemoteStore.java:176)
        at com.google.firebase.firestore.remote.WatchStream.onNext(WatchStream.java:108)
        at com.google.firebase.firestore.remote.WatchStream.onNext(WatchStream.java:38)
        at com.google.firebase.firestore.remote.AbstractStream$StreamObserver.lambda$onNext$1$AbstractStream$StreamObserver(AbstractStream.java:119)
        at com.google.firebase.firestore.remote.-$$Lambda$AbstractStream$StreamObserver$IZBnAkL3D_hQ5ofbDpSmX2lhuPo.run(Unknown Source:4)
        at com.google.firebase.firestore.remote.AbstractStream$CloseGuardedRunner.run(AbstractStream.java:67)
        at com.google.firebase.firestore.remote.AbstractStream$StreamObserver.onNext(AbstractStream.java:110)
        at com.google.firebase.firestore.remote.FirestoreChannel$1.onMessage(FirestoreChannel.java:125)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInternal(ClientCallImpl.java:658)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:643)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at com.google.firebase.firestore.util.AsyncQueue$SynchronizedShutdownAwareExecutor$DelayedStartFactory.run(AsyncQueue.java:229)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: io.grpc.StatusException: PERMISSION_DENIED: Missing or insufficient permissions.
        at io.grpc.Status.asException(Status.java:541)
        at com.google.firebase.firestore.util.Util.exceptionFromStatus(Util.java:115)
        at com.google.firebase.firestore.core.EventManager.onError(EventManager.java:166) 
        at com.google.firebase.firestore.core.SyncEngine.removeAndCleanupTarget(SyncEngine.java:588) 
        at com.google.firebase.firestore.core.SyncEngine.handleRejectedListen(SyncEngine.java:424) 
        at com.google.firebase.firestore.core.MemoryComponentProvider$RemoteStoreCallback.handleRejectedListen(MemoryComponentProvider.java:99) 
        at com.google.firebase.firestore.remote.RemoteStore.processTargetError(RemoteStore.java:562) 
        at com.google.firebase.firestore.remote.RemoteStore.handleWatchChange(RemoteStore.java:446) 
        at com.google.firebase.firestore.remote.RemoteStore.access$100(RemoteStore.java:53) 
        at com.google.firebase.firestore.remote.RemoteStore$1.onWatchChange(RemoteStore.java:176) 
        at com.google.firebase.firestore.remote.WatchStream.onNext(WatchStream.java:108) 
        at com.google.firebase.firestore.remote.WatchStream.onNext(WatchStream.java:38) 
        at com.google.firebase.firestore.remote.AbstractStream$StreamObserver.lambda$onNext$1$AbstractStream$StreamObserver(AbstractStream.java:119) 
        at com.google.firebase.firestore.remote.-$$Lambda$AbstractStream$StreamObserver$IZBnAkL3D_hQ5ofbDpSmX2lhuPo.run(Unknown Source:4) 
        at com.google.firebase.firestore.remote.AbstractStream$CloseGuardedRunner.run(AbstractStream.java:67) 
        at com.google.firebase.firestore.remote.AbstractStream$StreamObserver.onNext(AbstractStream.java:110) 
        at com.google.firebase.firestore.remote.FirestoreChannel$1.onMessage(FirestoreChannel.java:125) 
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInternal(ClientCallImpl.java:658) 
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:643) 
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) 
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at com.google.firebase.firestore.util.AsyncQueue$SynchronizedShutdownAwareExecutor$DelayedStartFactory.run(AsyncQueue.java:229) 
        at java.lang.Thread.run(Thread.java:764) 
```

Subsequent logins result in a white screen with no errors logged.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1032 

Skips the terms of service fragment if the document under "/config/tos" is missing in firestore.

Verified by removing the document from firebase and running the app locally.

@gino-m  PTAL?

### CODE REVIEW

**Comment by gino-m** [comment]
Hi Shobhit, Terms of Service should be optional, so there's no need to show a message or to quit. Would you be able to update this? Thanks again!

**Comment by shobhitagarwal1612** [comment]
> Hi Shobhit, Terms of Service should be optional, so there's no need to show a message or to quit. Would you be able to update this? Thanks again!

Done.

@gino-m Ready for another pass.

File: gnd/src/test/java/com/google/android/gnd/MainViewModelTest.java [hunk 1]
```diff
@@ -146,4 +146,19 @@ public void testSignInStateChanged_onSignInError() {
...
+    when(mockTosRepository.isTermsOfServiceAccepted()).thenReturn(false);
+    when(mockUserRepository.saveUser(any(User.class))).thenReturn(Completable.complete());
+    when(mockTosRepository.getTermsOfService()).thenReturn(Maybe.empty());
+
+    fakeAuthenticationManager.setUser(TEST_USER);
+    fakeAuthenticationManager.signIn();
+
+    assertProgressDialogVisible(false);
+    assertNavigate(HomeScreenFragmentDirections.showHomeScreen());
+    Mockito.verify(mockUserRepository, times(1)).saveUser(TEST_USER);
```
Comment (gino-m) [inline gnd/src/test/java/com/google/android/gnd/MainViewModelTest.java:161]: Nit: Should we static import these Mockito methods as well for consistency?

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks, Shobhit, especially for adding tests for this. One nit, otherwise LGTM.

File: gnd/src/test/java/com/google/android/gnd/MainViewModelTest.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply gnd/src/test/java/com/google/android/gnd/MainViewModelTest.java:161]: I've fixed this class in https://github.com/google/ground-android/pull/1031. Let's merge that first in order to fix here as well

Comment (gino-m) [reply gnd/src/test/java/com/google/android/gnd/MainViewModelTest.java:163]: Good idea. Merged, reran CI.

**Review by gino-m — DISMISSED** [review]
LGTM. Thanks for the fix!

**Comment by gino-m** [comment]
Looks like there are some missing imports in `MainViewModelTest`, possibly due to the merge?

**Review by gino-m — APPROVED** [review]

**Pull request merged**